server/src/models/mod.rs

```diff
@@ -252,6 +356,82 @@ mod tests {
+        let uncompressed_values = ValueArray::from(vec![10.0, 10.0, 10.0, 10.0, 10.0]);
+
+        let selected_model = create_selected_model(&uncompressed_timestamps, &uncompressed_values);
+
+        assert_eq!(PMC_MEAN_ID, selected_model.model_type_id);
+        assert_eq!(uncompressed_timestamps.len() - 1, selected_model.end_index);
+        assert_eq!(10.0, selected_model.min_value);
+        assert_eq!(10.0, selected_model.max_value);
+        assert_eq!(0, selected_model.values.len());
+    }
+
+    #[test]
+    fn test_model_selected_model_attributes_for_increasing_swing() {
+        let uncompressed_timestamps = TimestampArray::from_slice(UNCOMPRESSED_TIMESTAMPS);
+        let uncompressed_values = ValueArray::from(vec![10.0, 20.0, 30.0, 40.0, 50.0]);
+        let selected_model = create_selected_model(&uncompressed_timestamps, &uncompressed_values);
+
+        assert_eq!(SWING_ID, selected_model.model_type_id);
+        assert_eq!(uncompressed_timestamps.len() - 1, selected_model.end_index);
+        assert_eq!(10.0, selected_model.min_value);
+        assert_eq!(50.0, selected_model.max_value);
+        assert_eq!(0, selected_model.values.len());
+    }
+
+    #[test]
+    fn test_model_selected_model_attributes_for_decreasing_swing() {
+        let uncompressed_timestamps = TimestampArray::from_slice(UNCOMPRESSED_TIMESTAMPS);
+        let uncompressed_values = ValueArray::from(vec![50.0, 40.0, 30.0, 20.0, 10.0]);
+        let selected_model = create_selected_model(&uncompressed_timestamps, &uncompressed_values);
+
+        assert_eq!(SWING_ID, selected_model.model_type_id);
+        assert_eq!(uncompressed_timestamps.len() - 1, selected_model.end_index);
+        assert_eq!(10.0, selected_model.min_value);
+        assert_eq!(50.0, selected_model.max_value);
+        assert_eq!(0, selected_model.values.len());
+    }
+
+    #[test]
+    fn test_model_selected_model_attributes_for_gorilla() {
+        let uncompressed_timestamps = TimestampArray::from_slice(UNCOMPRESSED_TIMESTAMPS);
+        let uncompressed_values = ValueArray::from(vec![37.0, 73.0, 37.0, 73.0, 37.0]);
+        let selected_model = create_selected_model(&uncompressed_timestamps, &uncompressed_values);
+
+        assert_eq!(GORILLA_ID, selected_model.model_type_id);
+        assert_eq!(uncompressed_timestamps.len() - 1, selected_model.end_index);
+        assert_eq!(37.0, selected_model.min_value);
+        assert_eq!(73.0, selected_model.max_value);
+        assert_eq!(10, selected_model.values.len());
+    }
+
+    fn create_selected_model(
+        uncompressed_timestamps: &TimestampArray,
+        uncompressed_values: &ValueArray,
+    ) -> SelectedModel {
+        let error_bound = ErrorBound::try_new(0.0).unwrap();
+        let mut pmc_mean = PMCMean::new(error_bound);
+        let mut swing = Swing::new(error_bound);
+        let mut gorilla = Gorilla::new();
+
+        let mut pmc_mean_has_fit_all = true;
+        let mut swing_has_fit_all = true;
```

Comment:
Maybe change these to `...could_fit_all` to match the change to the struct fields.

Reply:
Good catch, fixed.